dask/dataframe/shuffle.py

```diff
@@ -562,22 +561,16 @@ def collect(p, part, meta, barrier_token):
 
 def set_partitions_pre(s, divisions):
     partitions = divisions.searchsorted(s, side="right") - 1
-    if is_series_like(partitions):
-        partitions = partitions.values
     partitions[(s >= divisions.iloc[-1]).values] = len(divisions) - 2
     return partitions
 
 
 def shuffle_group_2(df, col):
     if not len(df):
         return {}, df
-    ind = df[col].values.astype(np.int64)
+    ind = df[col].astype(np.int64)
     n = ind.max() + 1
-    indexer, locations = groupsort_indexer(ind.view(np.int64), n)
-    df2 = df.take(indexer)
-    locations = locations.cumsum()
-    parts = [df2.iloc[a:b] for a, b in zip(locations[:-1], locations[1:])]
-    result2 = dict(zip(range(n), parts))
+    result2 = methods.group_split(df, ind.values.view(np.int64), n)
```

Comment:
Why these changes?  Why call astype before `.values` and why the call to `view`?

Reply:
When `values` is a cupy array, `ind = df[col].values.astype(np.int64)` does not result in a scalar value, but in a cupy array with one element. I could explicitly convert this cupy array to a scalar inside of `group_split_cupy`, but the change I currently have avoids the need for this.

I'm not really sure about the use of `view` - It was already in the code.

Reply:
Ah indeed, I see now that view is there regardless.  

Ah, I see about cupy.max, so you wanted cudf/pandas to call max, because presumably they'll return scalars while numpy/cupy diverge here.  That seems reasonable to me.  I was just curious.

Another option would be to explicitly call `int`, which is probably fine in both cases.

```python
In [1]: import cupy

In [2]: x = cupy.arange(10)

In [3]: x.max()
Out[3]: array(9)

In [4]: int(x.max())
Out[4]: 9
```

But I can't see much benefit to this over what you have here.  So I say that we just leave this as you have it here.